build-data-capturing-gradle-samples/capture-test-pts-support/gradle-test-pts-support.gradle

```diff
@@ -1,76 +1,87 @@
+        'co.helmethair.scalatest.ScalatestEngine' : 'scalatest'
+    ]
+    private Logger logger
+
+    Capture(Logger logger) {
+        this.logger = logger
+    }
+
+    void capturePts(Test t, BuildScanExtension buildScanApi) {
+        if (t.getTestFramework().getClass().getName() == 'org.gradle.api.internal.tasks.testing.junitplatform.JUnitPlatformTestFramework') {
+            def engines = testEngines(t)
+            buildScanApi.value("${t.identityPath}#engines", "${engines}")
+            if (!engines.isEmpty() && engines.stream().allMatch { e -> supportedEngines.containsKey(e) }) {
+                buildScanApi.value("${t.identityPath}#pts", 'SUPPORTED')
             } else {
-                buildScanApi.value("${t.identityPath}#pts", 'NO_JUNIT_PLATFORM')
+                buildScanApi.value("${t.identityPath}#pts", 'ENGINES_NOT_ALL_SUPPORTED')
             }
+        } else {
+            buildScanApi.value("${t.identityPath}#pts", 'NO_JUNIT_PLATFORM')
         }
     }
-}
 
-Set<String> testEngines(Test t) {
-    try {
-        Stream<String> engines = t.classpath.files.stream()
+    Set<String> testEngines(Test t) {
+        try {
+            Stream<String> engines = t.classpath.files.stream()
                 .filter { f -> f.name.endsWith('.jar') }
                 .filter { f -> supportedEngines.values().stream().anyMatch { e -> f.name.contains(e) } }
                 .map { f -> findTestEngine(f) }
                 .flatMap { o -> o.isPresent() ? Stream.of(o.get()) : Stream.empty() }
 
-        // We take into account included/excluded engines (but only known ones)
-        def included = t.options.includeEngines
-        if (included) {
-            engines = engines.filter { e -> supportedEngines.get(e) == null || included.contains(supportedEngines.get(e)) }
+            // We take into account included/excluded engines (but only known ones)
+            def included = t.options.includeEngines
+            if (included) {
+                engines = engines.filter { e -> supportedEngines.get(e) == null || included.contains(supportedEngines.get(e)) }
+            }
+            def excluded = t.options.excludeEngines
+            if (excluded) {
+                engines = engines.filter { e -> supportedEngines.get(e) == null || !excluded.contains(supportedEngines.get(e)) }
+            }
+            return engines.collect(Collectors.toSet())
+        } catch (Exception e) {
+            logger.warn("Could not detect test engines", e)
         }
-        def excluded = t.options.excludeEngines  
-        if (excluded) {
-            engines = engines.filter { e -> supportedEngines.get(e) == null || !excluded.contains(supportedEngines.get(e)) }
-        }                  
-        return engines.collect(Collectors.toSet())
-    } catch (Exception e) {
-        gradle.rootProject.logger.warn("Could not detect test engines", e)
+        return Collections.emptySet()
```

Comment:
how come the previous code returned false but now this returns `Collections.emptySet()`? was the previous code not working in that case?

Reply:
yeah in Gradle it got unoticed because no compilation error and it's hard to test since it only triggers if there's an Exception thrown looking for test engines